src/hera_display_games/random_walk/__init__.py

```diff
@@ -2,31 +2,69 @@
 # -*- coding: utf-8 -*-
 """A simple simulation of a random walk"""
 import click
-from hera_display_games.core import mechanics
+from hera_display_games.core import mechanics, sprites, keymapper, map_dict
+import asyncio
 import time
 import random
 
 
+async def update_board(board, speed=10.0):
+    directions = ["r", "l", "dl", "dr", "ur", "ul"]
+    while True:
+        for sprite in board.sprites:
+            board.move_sprite(sprite, random.choice(directions))
+
+        board.draw()
+        await asyncio.sleep(1.0 / speed)
+
+
+# event loop and other code adapted from
+# https://github.com/AlexElvers/pygame-with-asyncio
+def pygame_event_loop(loop, event_queue):
+    import pygame
+
+    while True:
+        event = pygame.event.wait()
+        asyncio.run_coroutine_threadsafe(event_queue.put(event), loop=loop)
+
+
 @click.command()
 @click.option("-s", "--speed", type=float, default=1, help="Speed of sprite")
 @click.option("-n", "--nsprites", type=int, default=1, help="number of sprites")
+@click.option("-e/-E", "--eat/--rigid", default=True, help="whether sprites eat each other")
 @click.option(
     "-x",
     "-X",
     "--use-screen/--use-board",
     default=False,
     help="whether to use the physical HERA board",
 )
-def main(speed, nsprites, use_screen):
-    if not 1 <= nsprites <= 6:
-        raise ValueError("nsprites must be between 1 and 6")
+def main(speed, nsprites, eat, use_screen):
+    if not 1 <= nsprites <= 100:
+        raise ValueError("nsprites must be between 1 and 100")
 
     if not 1 <= speed <= 100:
         raise ValueError("speed must be between 1 and 100")
 
+    possible_pos = list(map_dict.reverse_led_map.keys())
+    positions = random.sample(possible_pos, nsprites)
+
+    cls = sprites.HungrySprite if eat else sprites.RigidSprite
     my_sprites = []
-    for i in range(nsprites):
-        my_sprites.append(mechanics.Sprite([0, i]))
+    for i, pos in enumerate(positions):
+        my_sprites.append(
+            cls(
+                list(map_dict.reverse_led_map[pos]),
+                color=(
+                    random.randint(100, 255),
+                    random.randint(100, 255),
+                    random.randint(100, 255),
```

Comment:
This is more a bookkeeping and choice kind of thing, but do we want to use the `random` module or `numpy.random`. There are already cases of both in the module, and I think for our specific needs neither is probably really more well optimized. I don't have strong feelings either way. I could see the argument for `random` just so `numpy` isn't required necessarily.

Reply:
I'd say `random` at least for now -- let's keep numpy out of it as long as possible.

Reply:
I'm fine with this. I'll make an issue to sort it out.